### ISSUE TITLE
Trouble Reproducing ArcFace Results

### ISSUE THREAD

**Issue by mdemoor226** [KevinMusgrave/powerful-benchmarker#94]
Hello there, thank you for making this useful piece of software. I am trying to reproduce some of the results in the Reality Check (Metric Learning) paper, but am having trouble getting the ArcFace Loss to converge properly. All the config settings are default (like in a fresh installation) with the exception of the Loss and Optimizer configs which are edited to use ArcFace with the optimal hyperparameters from the supplemental material. 

I've been able to reproduce the results of other losses (including CosFace) comfortably within the stated Confidence Intervals so I'm assuming this is due some bug (or user error I'm unaware of). Any help would be appreciated.

**Comment by KevinMusgrave** [comment]
Have you tried it multiple times?

**Comment by mdemoor226** [comment]
Pardon my poor wording I should have explained it better. What I meant was that I am unable to get ArcFace to train correctly at all. The Loss quickly drops to zero around the 4-5th epoch, but so does the Validation accuracy. This seems to happen for most hyperparameter combinations I've tried (including the optimal ones).  The strange part is CosFace runs just fine (which appears to share almost the entire codebase as ArcFace). I'm not sure why this is happening.

**Comment by KevinMusgrave** [comment]
I was able to reproduce the ArcFace accuracy using this notebook: https://colab.research.google.com/drive/1BAD_MKJTu5Plw3vENqhMo8NproSAB7kf?usp=sharing

I've attached the experiment logs from the above notebook
[experiments.zip](https://github.com/KevinMusgrave/powerful-benchmarker/files/8322715/experiments.zip)
d

If you run it in colab, then you have to upload the dataset yourself. Downloading the dataset doesn't work anymore 😩 

One thing you might have missed in your setup is ```--split_manager~SWAP~1 {MLRCSplitManager: {}}```. However, I don't think that should make much difference, because the default splitter and the MLRC splitter will both split the training set into 4 partitions. They just do it in slightly different ways.

**Comment by mdemoor226** [comment]
Thanks, that was indeed something I missed. I was also still using the default sampler. After swapping both using the arguments in the notebook, I was able to get the model to converge under ArcFace (on my own machine). However the accuracy was still slightly lower (getting about 19.7 mAP@R, 57.6 P@1, and 30.6 RP). 

This was on an Ubuntu 20.04 machine with 510 Nvidia Drivers, CUDA 11.3 and the latest version of PyTorch etc, everything installed via pip. So I decided to try everything on a different machine running CentOS in a conda environment and was able to reproduce the ArcFace results just fine. So it's either some weird portability problem or something I tweaked which I wasn't supposed to. I'll look into it again later when I can and update this if I find anything.
